src/main/java/org/truffleruby/core/fiber/FiberManager.java

```diff
@@ -143,31 +147,46 @@ private void fiberMain(RubyContext context, DynamicObject fiber, DynamicObject b
                 // Make sure that other fibers notice we are dead before they gain control back
                 Layouts.FIBER.setAlive(fiber, false);
             }
-            resume(fiber, Layouts.FIBER.getLastResumedByFiber(fiber), true, result);
+            resume(fiber, getReturnFiber(fiber, currentNode, UNPROFILED), FiberOperation.YIELD, result);
 
         // Handlers in the same order as in ThreadManager
         } catch (KillException e) {
             // Propagate the kill exception until it reaches the root Fiber
-            sendExceptionToParentFiber(fiber, e);
+            sendExceptionToParentFiber(fiber, e, currentNode);
```

Comment:
Could we catch some of these in the same block? E.g. `catch (KillException | ExitException e)` ?

Reply:
Yes, I just wanted to have it similar to the threadMain in ThreadManager, where there they have different handlers. But it makes sense to group them too.

Reply:
Done.